cpp/src/parquet/column_writer.cc

```diff
@@ -598,17 +589,18 @@ int64_t ColumnWriterImpl::Close() {
     FlushBufferedDataPages();
 
     EncodedStatistics chunk_statistics = GetChunkStatistics();
+    chunk_statistics.ApplyStatSizeLimits(
+        properties_->max_statistics_size(descr_->path()));
+    chunk_statistics.set_is_signed(SortOrder::SIGNED == descr_->sort_order());
+
     // Write stats only if the column has at least one row written
     // From parquet-mr
     // Don't write stats larger than the max size rather than truncating. The
     // rationale is that some engines may use the minimum value in the page as
     // the true minimum for aggregations and there is no way to mark that a
     // value has been truncated and is a lower bound and not in the page.
-    if (rows_written_ > 0 && chunk_statistics.is_set() &&
-        chunk_statistics.max_stat_length() <=
-            properties_->max_statistics_size(descr_->path())) {
```

Comment:
Checking for min, max stat lengths here previously was incorrect. We ended up not writing stats at all if the lengths exceeded the `max_statistics_size`. Now, the logic does the right thing by avoid only the `min`, `max` values but still setting `null` and `distinct` counts.